src/amo/reducers/addonsByAuthors.js

```diff
@@ -4,27 +4,31 @@ import invariant from 'invariant';
 
 import { createInternalAddon } from 'core/reducers/addons';
 import type {
-  SearchResultAddonType, ExternalAddonType,
+  AddonType, ExternalAddonType,
 } from 'core/types/addons';
 
 
-type State = {
+export type AddonsByAuthors = {|
   // TODO: It might be nice to eventually stop storing add-ons in this
   // reducer at all and rely on the add-ons in the `addons` reducer.
   // That said, these are partial add-ons returned from the search
   // results and fetching all add-on data for each add-on might be too
   // expensive.
-  byAddonId: { [number]: Array<SearchResultAddonType> },
+  byAddonId: { [number]: AddonType },
```

Comment:
Our convention in other reducers is to have `byId` store the object and `byX` store an ID. It's only that though: convention.

I don't wanna bikeshed it too much but because converting them would require going through several reducers and making changes I'd say we should leave it as-is but open a code quality issue about it. You make a fair point about clarifying that it's different.

Reply:
Great, thanks for explaining that. I think the convention is fine, if it is consistent across the repo.